### ISSUE TITLE
Integration needs reconfiguring

### ISSUE THREAD

**Issue by r00nster** [custom-components/alexa_media_player#1374]
Periodically (e.g. every 24hts) the integration needs reconfiguring and all entities become unavailable.

Installed the integration recently via HACS. All was working fine then a few days later the Integration showed as follows
![image](https://user-images.githubusercontent.com/40565100/136085754-f8707a3e-d8cd-4751-88f2-bee06ff3881c.png)

Couldn't get the integration to re-authenticate with Amazon so uninstalled, re-installed, reconfigured and all was fine again.

Then a couple days later I noticed the issue was back.

Is this normal behaviour (I expect not) ?

**System details**
 - Home-assistant (version): 2021.8.8
 - Installation Type: docker image
 - alexa_media  (version from `const.py` or HA startup): v3.10.8
 - alexapy (version from `pip show alexapy` or HA startup):
 - Amazon [2FA is enabled]: Yes

**Comment by kreene1987** [comment]
Agreed, here is the log entry. Looks to be OTP related:

`Error setting up entry k***@b**.com - amazon.com for alexa_media
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/alexapy/alexalogin.py", line 235, in set_totp
    self.get_totp_token()
  File "/usr/local/lib/python3.9/site-packages/alexapy/alexalogin.py", line 259, in get_totp_token
    token: Text = self._totp.now()
  File "/usr/local/lib/python3.9/site-packages/pyotp/totp.py", line 44, in now
    return self.generate_otp(self.timecode(datetime.datetime.now()))
  File "/usr/local/lib/python3.9/site-packages/pyotp/otp.py", line 33, in generate_otp
    hasher = hmac.new(self.byte_secret(), self.int_to_bytestring(input), self.digest)
  File "/usr/local/lib/python3.9/site-packages/pyotp/otp.py", line 50, in byte_secret
    return base64.b32decode(self.secret, casefold=True)
  File "/usr/local/lib/python3.9/base64.py", line 231, in b32decode
    raise binascii.Error('Non-base32 digit found') from None
binascii.Error: Non-base32 digit found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 304, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/alexa_media/__init__.py", line 303, in async_setup_entry
    AlexaLogin(
  File "/usr/local/lib/python3.9/site-packages/alexapy/alexalogin.py", line 119, in __init__
    self.set_totp(otp_secret.replace(" ", ""))
  File "/usr/local/lib/python3.9/site-packages/alexapy/alexalogin.py", line 241, in set_totp
    raise AlexapyPyotpInvalidKey(ex) from ex
alexapy.errors.AlexapyPyotpInvalidKey: Non-base32 digit found`

**Comment by whasf** [comment]
This started happening for me as well in the last few days. It went sooo long without having to re-authenticate (I'm talking at least 6 months), and now it needs it every day at least...

**Comment by Seechay** [comment]
I am having the same issue. I can't even remember the last time I had to re-authenticate, but the last few days have been crazy.

**Comment by alandtse** [comment]
Remove and readd the integration. Amazon had some issues recently as reported in #1373. 



> Agreed, here is the log entry. Looks to be OTP related:
> 
> >

Your issue appears different. The error is because whatever 2FA App key you're entering isn't a valid 2FA App key. Are you entering an OTP in the 2FA App key field? Regardless, your issue is off topic to the main issue. Please start a new issue for that.

**Comment by TomLyon** [comment]
I did remove the integration just the other day. Reinstalled and re-signed in, all was fine until today where the same exact issues occurred again. 

There's got to be more to it than just removing and re-adding the integration because that did not work for me.

**Comment by faspina** [comment]
Same thing, had it working after a delete and reinstall , now it came back and tells my my password is a problem

**Comment by martoq** [comment]
+1 for this issue.  Identical to those above.

**Comment by r00nster** [comment]
> I removed the integration and re-installed. All good now. Make sure you follow the install instructions for the new install method, you need the code GENERATOR ID, not the actual 6 digit code from the authenticator.
> 
> My issue is 100% resolved.

Can you explain how to do this pls ?

I've tried using the authenticator key and placing this in the integration, theis successful and I enter the generated code back on Amazon 2SV which is also successful and I get an new registered app. 

Then the integration says it must open a webpage and click the button. An Amazon logon page is display with username.password. I authenticate ok then get another OTP prompt.

At this stage no matter what OTP password I use (e.g. the original one from the integration or the one from the my authenticator app) I can't get past this.

**Comment by kreene1987** [comment]
Starting from scratch I entered my username and password, then on the 2FA I did this per the wiki:

Instead of using a separate 2FA code, you can also register a 2FA app with Alexa Media Player.

Go to Amazon.com's 2FA page and Add new App. You should save this key so you can reuse it if you reinstall.
Instead of scanning the QR code, select Can't scan the barcode.
Select the bolded value under Enter your Key (e.g., 35T5 LQSY I5IO 3EFQ LGAJ I6YB JWBY JJPR PYT7 XPPW IDAK SQBJ CVXA)
Enter the value from Step 3 in the Built-in 2FA App Key when adding the integration. 2FA should be automatically generated from now on and can be left blank.

Basically I gave HA the username, password, and 2FA code generator key, so if at any point and time it loses auth, it will regen a new key and reauth automatically.

Hopefully that helps?

**Comment by TomLyon** [comment]
Mine was fine again past few days then an hour ago I get this issue again. This has been a recurring issue for 2 weeks or so now.

**Comment by rmkew377** [comment]
I am Having this problem too. (Reauth required, then error)
In the configuration of the component the internal URI of HA is defined (192.168.x.x:8123) , but who uses it and for what reason?

**Comment by borsa93** [comment]
Hi all,
same issue here. I tried also proxy login method but is not working. how can I fix it? I followed exactly the wiki.
thanks all

**Comment by XxCabbyoO** [comment]
> 
> 
> Hi all, same issue here. I tried also proxy login method but is not working. how can I fix it? I followed exactly the wiki. thanks all

For me an homeassistant reboot works. After that i only have to enter the 2FA Code that comes automatically from Amazon via SMS after the homeassistant reboot.

**Comment by borsa93** [comment]
> > Hi all, same issue here. I tried also proxy login method but is not working. how can I fix it? I followed exactly the wiki. thanks all
> 
> For me an homeassistant reboot works. After that i only have to enter the 2FA Code that comes automatically from Amazon via SMS after the homeassistant reboot.

for me is working as well without reboot of HA. 
I need to enter the codes received by sms, but I think that we should find a way to not enter it again. thanks

**Comment by kreene1987** [comment]
I had to re-authenticate this morning, but by changing NOTHING it reauthenticated flawlessly. I'm wondering if the AMP team needs to look at programmed re-authentication on a certain timeline. Mine was almost exactly a month ago when last authenticated (28 days?).

Bummer that my music alarm didn't go off but the lights and stuff turned on since my alarm is HA-based and not AMP based so I still got up!

**Comment by emufan** [comment]
Same here. Since 4-5 days I have to reauth AMP every morning. I always get a SMS without requesting/re-adding/rebooting at 8 am CET. If I see this then later, this one is not valid anymore so I have to reboot HA, get a new SMS automatically with a new code and enter this and then it's good (but only till) 8 am CET next morning.

Didn't have this before. Any chance to get rid of this new re-auth every day, with all this reboots, etc.?
